tests/test_training.py

```diff
@@ -36,3 +37,31 @@ def test_training(max_steps: int = 100, threshold: float = 0.9):
                 break
 
         assert accuracy >= threshold, f"too small accuracy: {accuracy}"
+
+
+@pytest.mark.forked
+def test_decentralized_optimizer_step():
+    dht_root = DHT(start=True)
+    initial_peers = [f"127.0.0.1:{dht_root.port}"]
+
+    param1 = torch.nn.Parameter(torch.zeros(32, 32), requires_grad=True)
+    opt1 = DecentralizedOptimizer(opt=torch.optim.SGD([param1], lr=0.1),
+                                  dht=DHT(initial_peers=initial_peers, start=True),
+                                  prefix='foo', target_group_size=2, verbose=True)
+
+    param2 = torch.nn.Parameter(torch.ones(32, 32), requires_grad=True)
+    opt2 = DecentralizedOptimizer(opt=torch.optim.SGD([param2], lr=0.05),
+                                  dht=DHT(initial_peers=initial_peers, start=True),
+                                  prefix='foo', target_group_size=2, verbose=True)
```

Comment:
Also, pass the update frequency to these optimizers to ensure consistency with `time.sleep` below

Reply:
Alas, the time.sleep is meant not for consistency with averaging period, but simply to make sure that the slow CI executor was able to... well, do the averaging :)